sematic/examples/mnist/pytorch/pipeline.py

```diff
@@ -52,10 +57,19 @@ class PipelineConfig:
     use_cuda: bool = False
 
 
+# The node selector and toleration(s) you need to use to access
+# GPUs will vary depending on your Kubernetes setup.
 GPU_RESOURCE_REQS = ResourceRequirements(
     kubernetes=KubernetesResourceRequirements(
         node_selector={"node.kubernetes.io/instance-type": "g4dn.xlarge"},
         requests={"cpu": "2", "memory": "4Gi"},
+        tolerations=[
+            KubernetesToleration(
+                key="nvidia.com/gpu",
+                value="true",
+                effect=KubernetesTolerationEffect.NoSchedule,
+            )
+        ],
```

Comment:
Can we remove the `node_selector` above now? And K8s will select the appropriate node anyway?

Reply:
No, tolerations just say "I am allowed to run on nodes with a certain taint", it does not guarantee that the job will run on a job with that taint. The node selector guarantees we will have a node with a GPU. The GPU nodes want to "repel" jobs that don't require GPUs, so they have a taint. And then GPU jobs need a toleration to be able to ignore that taint.